netebpfext/net_ebpf_ext.c

```diff
@@ -227,6 +227,42 @@ static net_ebpf_ext_wfp_callout_state_t _net_ebpf_ext_wfp_callout_states[] = {
 // WFP globals
 static HANDLE _fwp_engine_handle;
 
+// We wait on this rundown during driver unload to ensure that we get 'filter deleted'
+// notifications from WFP for all our filters.  We acquire this rundown when we associate the
+// filter context with the first added filter and release it when we get the 'filter deleted'
+// notification for the last filter.
+typedef struct _net_ebpf_ext_filter_context_rundown
+{
+    EX_RUNDOWN_REF protection;
+} net_ebpf_filter_context_rundown_t;
+
+net_ebpf_filter_context_rundown_t _g_net_ebpf_filter_context_rundown;
+
+VOID
+net_ext_reference_filter_context(_In_opt_ net_ebpf_extension_wfp_filter_context_t* filter_context)
+{
+    if (filter_context != NULL) {
+        LONG _fc_local_ref_count = InterlockedIncrement(&filter_context->reference_count);
+        if (_fc_local_ref_count == 2) {
+            if (!ExAcquireRundownProtection(&_g_net_ebpf_filter_context_rundown.protection)) {
```

Comment:
Minor: I would not acquire here. I will acquire where the filter conext is created and ref count is set to 1.

Reply:
Yes. I did consider it.  Doing it here gives us the benefit of keeping all accesses to this specific rundown _within_ the ref/deref calls, which IMO provides better abstraction and also seems... cleaner, at least from a code readability perspective?